client/src/featureform/serving.py

```diff
@@ -188,23 +189,27 @@ def features(
         resp = self._stub.FeatureServe(req)
 
         feature_values = []
-        for val in resp.values:
-            parsed_value = parse_proto_value(val)
-            value_type = type(parsed_value)
-
-            # Ondemand features are returned as a byte array
-            # which holds the pickled function
-            if value_type == bytes:
-                code = dill.loads(bytearray(parsed_value))
-                func = types.FunctionType(code, globals(), "transformation")
-                parsed_value = func(self, params, entities)
-            # Vector features are returned as a Vector32 proto due
-            # to the inability to use the `repeated` keyword in
-            # in a `oneof` field
-            elif value_type == serving_pb2.Vector32:
-                parsed_value = parsed_value.value
 
-            feature_values.append(parsed_value)
+        for val_list in resp.values:
+            print(val_list)
+            entity_values = []
+            for val in val_list.values:
+                parsed_value = self.parse_proto_value(val)
+                value_type = type(parsed_value)
+
+                # Ondemand features are returned as a byte array
+                # which holds the pickled function
+                if value_type == bytes:
+                    code = dill.loads(bytearray(parsed_value))
+                    func = types.FunctionType(code, globals(), "transformation")
```

Comment:
does globals here mean any values globally stored will be part of the function execution? If true, souncds like a recipe for weird bugs?

Reply:
I don't think we're actually storing anything. This is what it was originally so I haven't modified it